core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockletIndexFactory.java

```diff
@@ -653,9 +653,13 @@ public SegmentProperties getSegmentPropertiesFromDataMap(Index coarseGrainIndex)
       throws IOException {
     List<Blocklet> blocklets = new ArrayList<>();
     List<CoarseGrainIndex> dataMaps = getIndexes(segment, partitions);
+    if (dataMaps.size() == 0 || dataMaps == null) {
```

Comment:
dataMaps cannot be null from getIndexes, it can be empty.
**So, remove it.**

Also If we checking null, it has to be the first condition of || else we can get NPE for null scenarios